cartographer_grpc/framework/server.h

```diff
@@ -55,12 +55,15 @@ class Server {
     void SetNumEventThreads(std::size_t num_event_threads);
     void SetServerAddress(const std::string& server_address);
 
-    template <typename RpcHandlerType, typename ServiceType>
-    void RegisterHandler(const std::string& method_name) {
-      std::stringstream fully_qualified_name;
-      fully_qualified_name << "/" << ServiceType::service_full_name() << "/"
-                           << method_name;
-      rpc_handlers_[ServiceType::service_full_name()].emplace(
+    template <typename RpcHandlerType>
+    void RegisterHandler() {
+      std::string method_full_name = GetMethodFullName<RpcHandlerType>();
+      std::string service_full_name;
+      std::string method_name;
+      std::tie(service_full_name, method_name) =
+          ParseMethodFullName(method_full_name);
+
```

Comment:
can you check that ServiceType::service_full_name() == service_full_name?

Reply:
Unfortunately not.